apps/studio/src/lib/data/tools.ts

```diff
@@ -9,26 +9,26 @@ function dec28bits(num: any): string {
 
 export const Mutators = {
 
-  resolveTabulatorMutator(dataType?: string): (v: any) => JsonFriendly {
-    const mutator = this.resolveDataMutator(dataType)
+  resolveTabulatorMutator(dataType?: string, conn: DBConnection = null): (v: any) => JsonFriendly {
+    const mutator = this.resolveDataMutator(dataType, conn)
     return (v: any) => mutator(v) // this cleans off the additional params
   },
 
-  resolveDataMutator(dataType?: string): (v: any, p?: boolean) => JsonFriendly {
+  resolveDataMutator(dataType?: string, conn: DBConnection = null): (v: any, p?: boolean) => JsonFriendly {
     if (dataType && dataType === 'bit(1)') {
       return this.bit1Mutator.bind(this)
     }
     if (dataType && dataType.startsWith('bit')) {
-      return this.bitMutator.bind(this)
+      return this.bitMutator.bind(this, conn)
     }
     return this.genericMutator.bind(this)
   },
 
 
-  mutateRow(row: any[], dataTypes: string[] = [], preserveComplex: boolean = false): JsonFriendly[] {
+  mutateRow(row: any[], dataTypes: string[] = [], preserveComplex: boolean = false, connection: DBConnection = null): JsonFriendly[] {
```

Comment:
I think you just need `dialect` here. See below. Also use `dialect?: Dialect` for optional